core/deployment/src/main/java/io/quarkus/deployment/IsDockerWorking.java

```diff
@@ -175,7 +175,7 @@ private static class DockerBinaryStrategy implements Strategy {
 
         private DockerBinaryStrategy(boolean silent) {
             this.silent = silent;
-            this.binary = ConfigProvider.getConfig().getOptionalValue("quarkus.docker.executable-name", String.class)
+            this.binary = ConfigProvider.getConfig().getOptionalValue("quarkus.native.container-runtime", String.class)
```

Comment:
Right, there is no good reason to have a different strategy. It might be a good time to have something common

Reply:
Note: The thing with `docker.executable-name` is that it's an option introduced in an extension, not in the core, so you might get somewhat confusing: 
```
[WARNING] [io.quarkus.config] Unrecognized configuration key "quarkus.docker.executable-name" 
was provided; it will be ignored; verify that the dependency extension for this configuration is set or
that you did not make a typo
```
I think that the whole "determine container runtime" should happen in the core (podman, docker, alias to one another) and both Test Framework and eventual extensions such as container-image-docker should just consume it.

The unlikely event of having to use one command to use builder image and another command to run containers could still be fine tuned by properties.

Not sure if this PR is the place where this whole thing gets refactored. I'll try.
I am less certain about the 2.16 branch, where going wild on refactoring all that matches word docker might be less desirable.